### PR TITLE
Vintage Story Models

### PR DESCRIPTION
**Pull request by Malik12tree**
Allows exporting Vintage Story models

### CODE REVIEW

**Comment by JannisX11** [comment]
Please adjust the plugin meta data. The plugin name is wrong, and the version should be semver (e. g. 1.0.0).

File: plugins/vintagestory_models.js [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+            })
+            MenuBar.addAction(exportVsAction, "file.export");
+        },
+        onunload() {
+            exportVsAction.delete();
+        }
+    });
+})()
+
+function getRangeBool(x, min, max) {
```
Comment (IanSSenne) [inline plugins/vintagestory_models.js:221]: these should prob be moved into the IIFE as to not crowd the global scope.

File: plugins.json [hunk 2]
```diff
@@ -380,5 +380,13 @@
 		"version": "1.0.0",
 		"min_version": "4.0.0",
 		"variant": "desktop"
+	},
+	"vintagestory_models": {
+		"title": "Reference Models",
+		"author": "Vintage Story Models",
+		"icon": "park",
+		"description": "Allows to export VintageStory models.",
+		"version": "1.0",
```
Comment (AzureDoom) [inline plugins.json:389]: You should be following https://semver.org/#summary so MAJOR.MINOR.FIX EX: 1.0.0

Comment (Malik12tree) [reply plugins.json]: ohh i didnt notice that, Thanks!!

**Comment by JannisX11** [comment]
Please test this on some models. I noticed an issue on a simple model I tried to export. For example, in line 46 you iterate over groups, but in the same function in 57 you are referring to cubes, which can cause a type error if there are more Groups than cubes.

**Comment by Malik12tree** [comment]
oh ok, I nearly tried on 4-5 different models and fixed some stuff, I think  **Now** everything should be done

**Pull request merged**